src/raftstore/store/fsm/peer.rs

```diff
@@ -1891,6 +1900,35 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         Some(ready_to_merge)
     }
 
+    fn on_ready_catch_up_logs(
+        &mut self,
+        merge: CommitMergeRequest,
+        logs_up_to_date: Arc<AtomicU64>,
+    ) {
+        let region_id = self.fsm.region_id();
+        assert_eq!(region_id, merge.get_source().get_id());
+        self.fsm.peer.catch_up_logs = Some(logs_up_to_date);
+
+        // directly append these logs to raft log
+        match self.fsm.peer.maybe_append_merge_entries(merge) {
+            Some(last_index) => {
+                info!(
+                    "append entries to source region";
+                    "region_id" => region_id,
+                    "peer_id" => self.fsm.peer.peer_id(),
+                    "last_index" => last_index,
+                );
```

Comment:
How about schedule a tick to the source region? It has some `Ready` committed entries.